sdk/typescript/src/sdk.ts

```diff
@@ -35,19 +67,33 @@ export class Pipeline {
         return this;
     }
 
+    private createPipeline(): PipelineSchema {
```

Comment:
Couple things on this one:

* Why not just use `schema.Schema` here, and delete `PipelineSchema`, which is essentially a duplicate? 
* Can we rename this to something that sounds less like instantiation? Something like `minify` might be more appropriate -- what do you think?

Reply:
1. The schema generation is a bit messy and while the generated `schema.Schema` is technically correct, there are type errors when trying to add a step to the schema interface. You can't add, for example, a `schema.CommandStep` as `schema.GroupStep`. So I thought it would be easier to just have a duplicate interface where things can be typed correctly instead of writing transformation functions for each step.

2. I've renamed it to `build` since the function is building/assembling the pipeline. Not sure `minify` makes sense because it isn't making anything smaller. The function is assembling the pipeline object.